test/expect/TestBatched.test_while.expect

```diff
@@ -11,33 +11,34 @@ graph(%a.1_data : Tensor
   %10 : Tensor = aten::__or__(%a.1_dims, %b_dims)
   %11 : int = prim::Constant[value=0]()
   %12 : Tensor = aten::mul(%8, %9)
-  %13 : Tensor = aten::sum(%12)
-  %14 : Tensor = aten::gt(%13, %11)
-  %15 : bool = prim::Bool(%14)
-  %16 : Tensor, %17 : Tensor, %a : Tensor, %19 : Tensor, %20 : Tensor = prim::Loop(%7, %15, %8, %9, %a.1_data, %a.1_mask, %a.1_dims)
+  %13 : int? = prim::None()
```

Comment:
This is not calling a cast, but adding a `prim::None` to `sum` symbol in line 15 below.

This is because this PR is removing the overloads without the dtypes, which is kind of the purpose of this PR.

Since the generated graphs include the default values from the schema, this will keep happening with every new kwarg added to ops. Two ways to fix this:
1. Keep overloads in `native_functions.yaml` and keep adding them with every new arg. (makes this PR unneeded)
2. Make JIT apply defaults at execution time. Maybe this behavior might be confined to kwargs only. (much larger scope than what this PR trying to achieve)